src/containerapp/azext_containerapp/tests/latest/utils.py

```diff
@@ -69,4 +72,224 @@ def create_and_verify_containerapp_up(
         # Re-run the 'az containerapp up' command with the location parameter if provided
         if location:
             up_cmd += f" -l {location.upper()}"
-            test_cls.cmd(up_cmd)
\ No newline at end of file
+            test_cls.cmd(up_cmd)
+
+def verify_containerapp_create_source_exception_without_ACR_registry_server(
```

Comment:
nit: it looks like the flow for the three exception verification functions you've added here are more-or-less the same except for the error message that's tracked -- is there any way we can consolidate the three exception verification functions into one that takes in all of the potential arguments, the exception that's expected and its message? That would cut down on some duplicate code quite a bit.

Reply:
Updated